From ddf1c09b90ac2dea503a0ef8d5e1f28ef366333e Mon Sep 17 00:00:00 2001
From: Yang Jihong <yangjihong1@huawei.com>
Date: Fri, 31 Dec 2021 15:59:38 +0800
Subject: [PATCH] kabi: reserve space for perf subsystem related structures
Patch-mainline: Not yet, from openEuler
References: bsn#22
openEuler-commit: ddf1c09b90ac2dea503a0ef8d5e1f28ef366333e
Modified-by-SEL: Yes, refreshed due to context changes


hulk inclusion
category: feature
bugzilla: https://gitee.com/openeuler/kernel/issues/I4GFVG?from=project-issue
CVE: NA

-------------------------------

Reserve space for the structures in perf subsystem.

Signed-off-by: Yang Jihong <yangjihong1@huawei.com>
Reviewed-by: Kuohai Xu <xukuohai@huawei.com>
Signed-off-by: Zheng Zengkai <zhengzengkai@huawei.com>
Signed-off-by: Guoqing Jiang <guoqing.jiang@suse.com>
---
 include/linux/perf_event.h | 8 ++++++++
 1 file changed, 8 insertions(+)

--- a/include/linux/perf_event.h
+++ b/include/linux/perf_event.h
@@ -57,6 +57,7 @@ struct perf_guest_info_callbacks {
 #include <linux/cgroup.h>
 #include <linux/refcount.h>
 #include <linux/security.h>
+#include <linux/kabi.h>
 #include <linux/static_call.h>
 #include <asm/local.h>
 
@@ -514,6 +515,13 @@ struct pmu {
 	 * Check period value for PERF_EVENT_IOC_PERIOD ioctl.
 	 */
 	int (*check_period)		(struct perf_event *event, u64 value); /* optional */
+
+	KABI_RESERVE(1)
+	KABI_RESERVE(2)
+	KABI_RESERVE(3)
+	KABI_RESERVE(4)
+	KABI_RESERVE(5)
+	KABI_RESERVE(6)
 };
 
 enum perf_addr_filter_action_t {
